shell/browser/api/electron_api_web_contents.cc

```diff
@@ -1418,14 +1432,6 @@ void WebContents::RequestMediaAccessPermission(
   permission_helper->RequestMediaAccessPermission(request, std::move(callback));
 }
 
-void WebContents::RequestToLockMouse(content::WebContents* web_contents,
-                                     bool user_gesture,
-                                     bool last_unlocked_by_target) {
-  auto* permission_helper =
-      WebContentsPermissionHelper::FromWebContents(web_contents);
-  permission_helper->RequestPointerLockPermission(user_gesture);
```

Comment:
It looks like this used to go through `WebContentsPermissionHelper::RequestPermission`, which I think triggered a JS event. The new code looks like it goes through some //chrome/ui stuff which I generally prefer to avoid.

Perhaps the existing permission event + the possibility of handling [Esc] through `before-input-event` is enough?

Thoughts?

Reply:
Looks like the old codepath went through ElectronPermissionManager::RequestPermissionsWithDetails(), which checks the handlers that were set via public API with `setPermissionRequestHandler()`.

I'm not sure what pointer is returned by `mouse_lock_controller()` -- @codebytere is it an ElectronPermissionManager s.t. user-provided request handlers are still reached here?

Reply:
it's the same exclusive access wiring that the keyboard lock uses, that was added here - https://github.com/electron/electron/pull/31572

Reply:
@codebytere fair enough, but let's make sure the app has a chance to cancel it!

Reply:
responded in a larger comment below!